### PR TITLE
Issue 9: Standardize all private keys to OneAssymetricKey format and public keys to SubjectPublicKeyInfo format

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

**Review by mhabrat — CHANGES_REQUESTED** [review]

File: reference/src/taimpl/src/internal/typej.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2019-2021 Comcast Cable Communications Management, LLC
+ * Copyright 2019-2022 Comcast Cable Communications Management, LLC
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/taimpl/src/internal/typej.c:2]: I forgot to change it when I completed Issue 3.

File: reference/src/client/src/sa_engine_pkey.c [hunk 2]
```diff
@@ -1287,15 +1253,171 @@ static EVP_PKEY_METHOD* get_pkey_method(
...
+                break;
+            }
+
+            OSSL_PARAM params[] = {
+                    OSSL_PARAM_construct_utf8_string(OSSL_PKEY_PARAM_GROUP_NAME, group_name, group_name_length),
+                    OSSL_PARAM_construct_end()};
+
+            ec_group = EC_GROUP_new_from_params(params, NULL, NULL);
+            if (ec_group == NULL) {
+                ERROR("EVP_PKEY_get_octet_string_param failed");
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/src/sa_engine_pkey.c:1337]: Done

File: reference/src/client/test/sa_engine_pkey_decrypt.cpp [hunk 3]
```diff
@@ -63,13 +63,24 @@ TEST_P(SaEnginePkeyEncryptTest, encryptTest) {
...
+                free(new_label);
+                GTEST_FATAL_FAILURE_("EVP_PKEY_CTX_ctrl failed");
+            }
 #else
-            ASSERT_EQ(EVP_PKEY_CTX_set0_rsa_oaep_label(encrypt_pkey_ctx.get(), new_label, label.size()), 1);
+
+            if (EVP_PKEY_CTX_set0_rsa_oaep_label(encrypt_pkey_ctx.get(), new_label,
+                        static_cast<int>(label.size())) != 1) {
+                free(new_label);
+                GTEST_FATAL_FAILURE_("EVP_PKEY_CTX_ctrl failed");
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/test/sa_engine_pkey_decrypt.cpp:82]: Done

File: reference/src/client/test/sa_engine_pkey_decrypt.cpp [hunk 4]
```diff
@@ -93,13 +104,23 @@ TEST_P(SaEnginePkeyEncryptTest, encryptTest) {
...
+                        EVP_PKEY_CTRL_RSA_OAEP_LABEL, static_cast<int>(label.size()), new_label) != 1) {
+                free(new_label);
+                GTEST_FATAL_FAILURE_("EVP_PKEY_CTX_ctrl failed");
+            }
 #else
-            ASSERT_EQ(EVP_PKEY_CTX_set0_rsa_oaep_label(decrypt_pkey_ctx.get(), new_label, label.size()), 1);
+            if (EVP_PKEY_CTX_set0_rsa_oaep_label(decrypt_pkey_ctx.get(), new_label,
+                        static_cast<int>(label.size())) != 1) {
+                free(new_label);
+                GTEST_FATAL_FAILURE_("EVP_PKEY_CTX_ctrl failed");
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/test/sa_engine_pkey_decrypt.cpp:122]: Done

**Review by mhabrat — APPROVED** [review]

**Pull request merged**